test/node-test/abort-controller.js

```diff
@@ -19,87 +90,104 @@ if (global.AbortController) {
   })
 }
 for (const { AbortControllerImpl, controllerName } of controllers) {
-  test(`Abort ${controllerName} before creating request`, (t) => {
-    t.plan(1)
+  test(`Abort ${controllerName} before creating request`, async (t) => {
+    const p = tspl(t, { plan: 1 })
 
     const server = createServer((req, res) => {
-      t.fail()
+      p.fail()
     })
-    t.teardown(server.close.bind(server))
+    t.after(server.close.bind(server))
+
+    const { promise, resolve } = promiseWithResolvers()
```

Comment:
How can I get rid of promise waiting hacks like this...?

Reply:
I found out that all I need to do is await p.completed. I missed it because the type definition was wrong. ( https://github.com/mcollina/tspl/pull/3 )